src/csharp/Microsoft.Spark.Worker.UnitTest/SimpleWorkerTests.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Spark.Network;
+using Xunit;
+
+namespace Microsoft.Spark.Worker.UnitTest
+{
+    public class SimpleWorkerTests
+    {
+        [Fact]
+        public void TestsSimpleWorkerTaskRunners()
+        {
+            var typedVersion = new Version(Versions.V2_4_0);
+            var simpleWorker = new SimpleWorker(typedVersion);
+            ISocketWrapper serverListen = SocketFactory.CreateSocket();
+            var ipEndpoint = (IPEndPoint)serverListen.LocalEndPoint;
+            int port = ipEndpoint.Port;
+
+            serverListen.Listen();
+
+            PayloadWriter payloadWriter = new PayloadWriterFactory().Create();
+            Task clientTask = Task.Run(() => simpleWorker.Run(port));
+
+            using (ISocketWrapper serverSocket = serverListen.Accept())
```

Comment:
Can we merge with TaskRunnerTests which has the same test?

Reply:
sure! I'm working on it now

Reply:
resolved at 4ececb7